datumaro/plugins/pointcloud_format/converter.py

```diff
@@ -0,0 +1,302 @@
+
+                    label_object = {
+                        "key": self.get_object_key(item.id),
+                        "classTitle": item.name,
+                        "tags": [],
+                        "labelerLogin": self._user["name"],
+                        "createdAt": str(self._user["createdAt"]),
+                        "updatedAt": str(self._user["updatedAt"])
+                    }
+                    self._meta_data["classes"].append(classes)
+                    self._label_objects.append(label_object)
+
+    def generate_frames(self):
+        for i, data in enumerate(self._annotation):
+            frame_data = []
+            if data.pcd:
+                self._write_item(data, data.id)
+                if not self.get_video_key(int(data.attributes['frame'])):
+                    self.set_videos_key(int(data.attributes["frame"]))
+
+            for item in data.annotations:
+
+                if item.type == AnnotationType.cuboid:
+
+                    self.set_figures_key(item.id)
+                    figures = {
+                        "key": self.get_figure_key(item.id),
+                        "objectKey": self.get_object_key(item.attributes["label_id"]),
+                        'geometryType': "cuboid_3d",
+                        "geometry": {
+                            "position": {
+                                "x": item.points[0],
+                                "y": item.points[1],
+                                "z": item.points[2]
+                            },
+                            "rotation": {
+                                "x": item.points[3],
+                                "y": item.points[4],
+                                "z": item.points[5]
+                            },
+                            "dimensions": {
+                                "x": item.points[6],
+                                "y": item.points[7],
+                                "z": item.points[8]
+                            }
+                        },
+                        "labelerLogin": self._user["name"],
+                        "createdAt": str(self._user["createdAt"]),
+                        "updatedAt": str(self._user["updatedAt"])
+                    }
+                    frame_data.append(figures)
+            if frame_data:
+                self._frame_data[int(data.attributes["frame"])] = frame_data
+
+    def _write_item(self, item, index):
+        if not self._context._reindex:
+            index = cast(item.attributes.get('frame'), int, index)
+        image_info = OrderedDict([("id", str(index)), ])
+        if isinstance(item.pcd, str) and osp.isfile(item.pcd):
+            path = item.pcd.replace(os.sep, '/')
+            filename= path.rsplit("/", maxsplit=1)[-1]
+        else:
+            filename = self._context._make_pcd_filename(item)
+        image_info["name"] = filename
+        PointCloudPath.WRITE_FILES.append({index: filename})
```

Comment:
I think, changing a global variable here is a bad idea.

Reply:
Resolved